public/app/features/scenes/scenes/nestedBasic.tsx

```diff
@@ -0,0 +1,93 @@
+import { getDefaultTimeRange } from '@grafana/data';
+
+import { NestedScene } from '../components/NestedScene';
+import { Scene } from '../components/Scene';
+import { SceneFlexChild, SceneFlexLayout } from '../components/SceneFlexLayout';
+import { SceneTimePicker } from '../components/SceneTimePicker';
+import { VizPanel } from '../components/VizPanel';
+import { SceneDataProviderNode } from '../core/SceneDataProviderNode';
+import { SceneTimeRange } from '../core/SceneTimeRange';
+import { SceneEditManager } from '../editor/SceneEditManager';
+
+export function getScene(): Scene {
+  const scene = new Scene({
+    title: 'Nested Scene demo (isolated)',
+    $editor: new SceneEditManager({}),
+    children: [
+      new SceneFlexLayout({
+        direction: 'column',
+        children: [getInnerScene('Inner scene')],
+      }),
+    ],
+  });
+
+  return scene;
+}
+
+export function getInnerScene(title: string) {
+  const timeRangeNode = new SceneTimeRange({
+    range: getDefaultTimeRange(),
+  });
+
+  const dataProviderNode = new SceneDataProviderNode({
+    inputParams: {
+      timeRange: timeRangeNode,
+    },
+    queries: [
+      {
+        refId: 'A',
+        datasource: {
+          uid: 'gdev-testdata',
+          type: 'testdata',
+        },
+        scenarioId: 'random_walk_table',
+      },
+    ],
+  });
+
+  const scene = new NestedScene({
```

Comment:
I found confusing the naming for nested scenes. I was assuming nested scenes were Scene instances, but they are just SceneObjects with an specific behaviour. I'd reconsider the name as something else not related to Scene. 

IMO, conceptually it make sense to have an scenes under another scene, maybe we can reuse the Scene instance to make it composable as well. Just an idea